File: c/chapitre_3/exo33.c
/*
Ecrire un programme qui saisit un nom de ﬁchier texte au clavier et ouvre ce
ﬁchier dans l’éditeur emacs, dont le ﬁchier exécutable se trouve à l’emplacement /usr/bin/emacs.
*/

#include <stdio.h>
#include <unistd.h>
#include <stdlib.h>

int main()
{
    char *fichier ;

    printf("Entrez le fichier à éditer : ");
    scanf("%s",fichier) ;

    execl("/usr/bin/emacs" , "emacs" , fichier , NULL ) ;

    return 0 ;
}

File: python/chapitre_3/exo33.py
"""
Ecrire un programme qui saisit un nom de ﬁchier texte au clavier et ouvre ce
ﬁchier dans l'éditeur emacs.
"""

import os

def main():
    fichier = input("Entrez le fichier à éditer : ")
    os.system(f"emacs {fichier}")

if __name__ == "__main__":
    main()